cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sources/KafkaSource.java

```diff
@@ -108,11 +105,12 @@ public SourceState poll(Emitter<StructuredRecord> writer, SourceState currentSta
    * @param payload the ByteBuffer of the Kafka message.
    * @return instance of {@link StructuredRecord} representing the message.
    */
-  public StructuredRecord byteBufferToStructuredRecord(@Nullable String key, ByteBuffer payload) {
+  public StructuredRecord byteBufferToStructuredRecord(@Nullable String key, byte[] payload) {
+    StructuredRecord.Builder recordBuilder = StructuredRecord.builder(SCHEMA);
     if (key != null) {
       recordBuilder.set(KEY, key);
     }
-    recordBuilder.set(MESSAGE, payload.array());
```

Comment:
Yes, I should not have used ByteBuffer.array to set the message field. 
That is why I was thinking about sending the byte array instead of ByteBuffer to simplify the downstream consumption of KafkaSource output.